Object Store/func.py

```diff
@@ -1,67 +1,85 @@
-        obj = data.get("resourceName")
-        eventtime = body.get("eventTime")
+    except (Exception, ValueError) as ex:
+        logging.getLogger().info(str(ex))
+        return
 
-        source = "Oracle Cloud" #adding a source name.
-        service = "OCI Logs" #adding a servicen name.
+    data = body.get("data", {})
+    additional_details = data.get("additionalDetails", {})
 
-        datafile = request_one_object(namespace, bucket, obj)
-        data = str(datafile,'utf-8')
+    namespace = additional_details.get("namespace")
+    if not namespace:
+        logging.getLogger().error("No namespace provided")
+        return
+
+    bucket = additional_details.get("bucketName")
+    if not bucket:
+        logging.getLogger().error("No bucket provided")
+        return
+
+    resource_name = data.get("resourceName")
+    if not resource_name:
+        logging.getLogger().error("No obj provided")
+        return
 
-        datadoghost = os.environ['DATADOG_HOST']
-        datadogtoken = os.environ['DATADOG_TOKEN']
+    event_time = body.get("eventTime")
 
-        for lines in data.splitlines():
-            logging.getLogger().info("lines " + lines)
-            payload = {}
-            payload.update({"host":obj})
-            payload.update({"time": eventtime})
+    source = "Oracle Cloud"  # Adding a source name.
+    service = "OCI Logs"     # Adding a service name.
 
-            payload.update({"ddsource":source}) 
-            payload.update({"service":service})
+    datafile = request_one_object(namespace, bucket, resource_name)
+    data = str(datafile, 'utf-8')
 
-            payload.update({"event":lines})
+    # Datadog endpoint URL and token to call the REST interface.
+    # These are defined in the func.yaml file.
+    try:
+        dd_host = os.environ['DATADOG_HOST']
+        dd_token = os.environ['DATADOG_TOKEN']
+        dd_tags = os.environ.get('DATADOG_TAGS', '')
+    except KeyError:
+        err_msg = "Could not find environment variables, \
+                   please ensure DATADOG_HOST and DATADOG_TOKEN \
+                   are set as environment variables."
+        logging.getLogger().error(err_msg)
 
-            
- 
-        headers = {'Content-type': 'application/json', 'DD-API-KEY': datadogtoken}
-        x = requests.post(datadoghost, data = json.dumps(payload), headers=headers)
-        logging.getLogger().info(x.text)
-        print(x.text)
+    for lines in data.splitlines():
+        logging.getLogger().info("lines %s", lines)
+        payload = {}
+        payload.update({"ddsource": source})
+        payload.update({"ddtags": dd_tags})
+        payload.update({"host": resource_name})
+        payload.update({"time": event_time})
+        payload.update({"service": service})
+        payload.update({"event": lines})
 
+    try:
+        headers = {'Content-type': 'application/json', 'DD-API-KEY': dd_token}
+        req = requests.post(dd_host, data=json.dumps(payload), headers=headers)
```

Comment:
nit: should `req` be initialized if not using the variable. If using this, should `res`/`response` be a better option?